### DISCUSSION TITLE
Creating .NET MAUI projects

### DISCUSSION THREAD

**Discussion by nikolavn** [microsoft/vscode-discussions#898] [Extension Development QnA] [answered]
I am looking for information about the appropriate approach for handling templates within Visual Studio Code, particularly concerning the creation of .NET MAUI projects. Presently, we are using templates to generate such projects in Visual Studio for Mac. However, as it will be retired soon, our aim is to transition these processes to Visual Studio Code.

While exploring alternatives, we examined the WindowsTemplateStudio (we are already using WebTemplateStudio for other projects) but it doesn't offer .NET MAUI templates. Furthermore, it appears that no equivalent to VisualStudio.TemplateEngine is available for Visual Studio Code.

I am looking for any recommended practices for the initiation of these templates within Visual Studio Code without solely relying on launching the dotnet process in the background and passing the requisite parameters.

**Comment by isidorn** [comment, marked as answer]
I suggest to file an issue here https://github.com/microsoft/vscode-dotnettools/issues
And our dotnet friends should be able to help. Thank you 🙏

**Reply by nikolavn** [reply]
Thanks, I did that. [Here](https://github.com/microsoft/vscode-dotnettools/issues/694) is the issue for anyone interested.
